Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -1486,7 +1488,16 @@ void CSSParserImpl::consumeBlockContent(CSSParserTokenRange range, StyleRuleType
                 RefPtr rule = consumeAtRule(range, AllowedRules::RegularRules);
                 if (!rule)
                     break;
-                if (!rule->isGroupRule())
+                // Weirdly, @scope allows global name-defining rule, but not inside style rule.
+                // https://drafts.csswg.org/css-cascade-6/#scope-scope
+                auto hasAncestorStyleRule = [&] {
+                    for (const auto& rule : m_ancestorRuleTypeStack) {
+                        if (rule == CSSParserEnum::NestedContextType::Style)
+                            return true;
+                    }
+                    return false;
+                }();
+                if (hasAncestorStyleRule && !rule->isGroupRule())
```

Comment:
Is this a hot code path? Maybe it would be faster to always check `!rule->isGroupRule()` before iterating over the rule type stack since we can fast-fail that way?

Reply:
Good idea it could be potentially hot. Changing the code to use short-circuiting